WearTilesKotlin/app/src/debug/java/com/example/wear/tiles/TilesKotlinScreen.kt

```diff
@@ -49,6 +62,7 @@ internal fun SampleTilesList(context: Context, sampleTiles: Map<Int, Class<out T
             }
         }
     }
+    FocusOnResume(focusRequester = focusRequester)
```

Comment:
It would help me if you could comment on each statement with confirmation/clarifications 🙏 

1. a composable (like this screen) can/should request focus for descendants, like the SLC. It's not appropriate for SLC to request focus on itself
2. here, the screen is not requesting focus immediately, it's using lifecycle callbacks to _then_ request focus. Are there cases where requesting focus immediately (i.e. `focusRequester.requestFocus()`) makes sense?
3. We're passing `focusRequester` to the `scrollableColumn` modifier above. This is not used to request focus. It's so that we (the scrollableColumn modifier) can be notified about changes to focus.

`#3` is the most difficult one to wrap my head around, presumably because I don't understand how Compose is working with these events.

Reply:
1) Yes. An individual component can't know that it should be the requested item to start with. There can be multiple on a single screen, think of a DatePicker.  Or there can be multiple screens composed at same time.  Think HorizontalPager, SwipeToDismissBox or SwipeDismissableNavHost.

2) I'm not 100% definitive on the definitions. My understanding is focusable, makes the item focusable through general focus navigation, e.g. dtab, or textfields. focusRequester is a controller that lets you do this programmatically.

See https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/ui/ui/samples/src/main/java/androidx/compose/ui/samples/FocusSamples.kt;l=103?q=RequestFocusSample

It's hard to think about cases where immediately requesting focus makes sense, since you don't know if you are composed before you ever come on screen if at all.

3) I think answered in 2. I don't think the focusRequester is about getting notified of focus, I think it's about programmatically requesting focus in a UI toolkit that doesn't have 30 fields of views that you can call methods on.

Reply:
nice, I'll play with that sample 🙌  thank you